lib/vagrant-libvirt/action.rb

```diff
@@ -384,6 +390,51 @@ def self.action_ssh_run
         end
       end
 
+      # Snapshot actions
+      def self.action_snapshot_delete
+        Vagrant::Action::Builder.new.tap do |b|
+          b.use ConfigValidate
+          b.use Call, IsCreated do |env, b2|
+            unless env[:result]
+              raise Vagrant::Errors::VMNotCreatedError
+            end
+
+            b2.use SnapshotDelete
+          end
+        end
+      end
+      
+      # This is the action that is primarily responsible for restoring a snapshot
+      def self.action_snapshot_restore
+        Vagrant::Action::Builder.new.tap do |b|
+          b.use ConfigValidate
+          b.use Call, IsCreated do |env, b2|
+            unless env[:result]
+              raise Vagrant::Errors::VMNotCreatedError
+            end
+            
+            # Using as an example the virtualbox plugin, we don't delete the
+            # snapshot after restoring it 
+            # https://github.com/hashicorp/vagrant/blob/main/plugins/providers/virtualbox/action/snapshot_restore.rb
+            b2.use SnapshotRestore
```

Comment:
In the original PR, #840, there was a comment questioning that if it's really necessary to delete the snapshots after restoring then. 

I removed that piece of code here because following the example of the virtualbox plugin in vagrant repository, they don't delete the snapshot after restoring them. 

Link to comment: https://github.com/vagrant-libvirt/vagrant-libvirt/pull/840#discussion_r156145695
Link to the snapshot_restore file in virtualbox plugin: https://github.com/hashicorp/vagrant/blob/main/plugins/providers/virtualbox/action/snapshot_restore.rb

Reply:
yeah, it looked strange to me to delete the snapshot after a restore (thought about what if the user wants to restore the snapshot multiple times? It couldn't in the old PR as it would be deleted after the first one).